src/de/muenchen/allg/itd51/wollmux/event/WollMuxEventHandler.java

```diff
@@ -1,4987 +1,1082 @@
-        lock[0] = false;
-        lock.notifyAll();
-      }
-      enableAllOOoWindows(true);
-    }
-
-    /**
-     * Wartet so lange, bis der vorher mit setLock() gesetzt lock mit der Methode
-     * setUnlock() aufgehoben wird. So kann die quasi Modalität nicht modale Dialoge
-     * zu realisiert werden. setLock() sollte stets vor dem Aufruf des nicht modalen
-     * Dialogs erfolgen, nach dem Aufruf des nicht modalen Dialogs folgt der Aufruf
-     * der waitForUnlock()-Methode. Der nicht modale Dialog erzeugt bei der
-     * Beendigung ein ActionEvent, das dafür sorgt, dass setUnlock aufgerufen wird.
-     */
-    protected void waitForUnlock()
-    {
-      try
-      {
-        synchronized (lock)
-        {
-          while (lock[0] == true)
-            lock.wait();
-        }
-      }
-      catch (InterruptedException e)
-      {}
-    }
-
-    /**
-     * Dieser ActionListener kann nicht modalen Dialogen übergeben werden und sorgt
-     * in Verbindung mit den Methoden setLock() und waitForUnlock() dafür, dass quasi
-     * modale Dialoge realisiert werden können.
-     */
-    protected UnlockActionListener unlockActionListener = new UnlockActionListener();
-
-    protected class UnlockActionListener implements ActionListener
-    {
-      public ActionEvent actionEvent = null;
-
-      @Override
-      public void actionPerformed(ActionEvent arg0)
-      {
-        actionEvent = arg0;
-        setUnlock();
-      }
-    }
-  }
-
-  /**
-   * Stellt das WollMuxEvent event in die EventQueue des EventProcessors.
-   *
-   * @param event
-   */
-  private static void handle(WollMuxEvent event)
-  {
-    EventProcessor.getInstance().addEvent(event);
-  }
-
-  // *******************************************************************************************
-
-  /**
-   * Erzeugt ein neues WollMuxEvent, das den Dialog AbsenderAuswaehlen startet.
-   */
-  public static void handleShowDialogAbsenderAuswaehlen()
-  {
-    handle(new OnShowDialogAbsenderAuswaehlen());
-  }
-
-  /**
-   * Dieses Event wird vom WollMux-Service (...comp.WollMux) und aus dem
-   * WollMuxEventHandler ausgelöst und sorgt dafür, dass der Dialog AbsenderAuswählen
-   * gestartet wird.
-   *
-   * @author christoph.lutz
-   */
-  private static class OnShowDialogAbsenderAuswaehlen extends BasicEvent
-  {
-    @Override
-    protected void doit() throws WollMuxFehlerException
-    {
-      ConfigThingy conf = WollMuxFiles.getWollmuxConf();
-
-      try
-      {
-        // Konfiguration auslesen:
-        ConfigThingy whoAmIconf = requireLastSection(conf, "AbsenderAuswaehlen");
-        ConfigThingy PALconf = requireLastSection(conf, "PersoenlicheAbsenderliste");
-        ConfigThingy ADBconf = requireLastSection(conf, "AbsenderdatenBearbeiten");
-
-        // Dialog modal starten:
-        setLock();
-        new AbsenderAuswaehlen(whoAmIconf, PALconf, ADBconf,
-          DatasourceJoinerFactory.getDatasourceJoiner(), unlockActionListener);
-        waitForUnlock();
```

Comment:
Ist das noch die richtige Stelle für die Methode?

Reply:
der Code befindet sich inzwischen in der Klasse OnShowDialogAbsenderAuswaehlen